processor/k8sattributesprocessor/internal/kube/kube.go

```diff
@@ -270,6 +270,19 @@ func (r *FieldExtractionRule) extractField(v string) string {
 	return ""
 }
 
+// RegexpCompile make pattern fully anchored
+func RegexpCompile(pattern string) (*regexp.Regexp, error) {
+	return regexp.Compile("^(?:" + pattern + ")$")
```

Comment:
Why the non-capturing group is needed here? Doesn't just `("^" + pattern + "$")` work?

Reply:
Beacause isn't safe, please sonsider `a|b` will match `accc` incorrectly. In fact,  it should mean match `a or b` instead of mean start with `a` or end with `b`

Reply:
Oh I see. Good point!